### PR TITLE
Spearbit 83

### PR DESCRIPTION
**Pull request by liu-zhipeng**
## Description
Fix https://github.com/spearbit-audits/connext-nxtp/issues/83

## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [x] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

Fixes 
https://github.com/spearbit-audits/connext-nxtp/issues/83

### CODE REVIEW

File: packages/deployments/contracts/contracts/core/connext/libraries/AmplificationUtils.sol [hunk 1]
```diff
@@ -44,24 +44,18 @@ library AmplificationUtils {
...
       uint256 t0 = self.initialATime; // time when ramp is started
-      uint256 a0 = self.initialA; // initial A value when ramp is started
-      if (a1 > a0) {
-        // a0 + (a1 - a0) * (block.timestamp - t0) / (t1 - t0)
-        return a0 + ((a1 - a0) * (block.timestamp - t0)) / (t1 - t0);
-      } else {
-        // a0 - (a0 - a1) * (block.timestamp - t0) / (t1 - t0)
-        return a0 - ((a0 - a1) * (block.timestamp - t0)) / (t1 - t0);
+      assembly {
+        currentA := div(add(mul(a0, sub(t1, timestamp())), mul(currentA, sub(timestamp(), t0))), sub(t1, t0))
```
Comment (Saw-mon-and-Natalie) [inline packages/deployments/contracts/contracts/core/connext/libraries/AmplificationUtils.sol:57]: This is a long line can be formatted into:

```solidity
currentA := div(
   add(
      mul(a0, sub(t1, timestamp())),
      mul(currentA, sub(timestamp(), t0))
   ), 
   sub(t1, t0)
)
```

**Review by LayneHaber — CHANGES_REQUESTED** [review]
check assumptions, formatting

File: packages/deployments/contracts/contracts/core/connext/libraries/AmplificationUtils.sol (hunk 1 above)
Comment (liu-zhipeng) [reply packages/deployments/contracts/contracts/core/connext/libraries/AmplificationUtils.sol:57]: Solidity prettier made so automatically.

Comment (LayneHaber) [reply packages/deployments/contracts/contracts/core/connext/libraries/AmplificationUtils.sol:57]: lets just leave it -- changing prettier would be a huge whitespace diff on every PR rn

**Review by LayneHaber — APPROVED** [review]
seems assumptions outlined are being checked!

**Comment by Saw-mon-and-Natalie** [comment]
[`_getAPrecise`](https://github.com/connext/nxtp/blob/f168866ca4df13be4bab814e0a2d2578bf448224/packages/deployments/contracts/contracts/core/connext/libraries/SwapUtilsExternal.sol#L152-L169) in `SwapUtilsExternal` can also be updated .

**Comment by liu-zhipeng** [comment]
> [`_getAPrecise`](https://github.com/connext/nxtp/blob/f168866ca4df13be4bab814e0a2d2578bf448224/packages/deployments/contracts/contracts/core/connext/libraries/SwapUtilsExternal.sol#L152-L169) in `SwapUtilsExternal` can also be updated .

Fixed!

**Review by Saw-mon-and-Natalie — APPROVED** [review]
Looks good. About the assumption that `block.timestamp` would need to be always greater than or equal to  `t0 = initialATime`. It is also true on initialization since we start with `t0 = 0`.

**Pull request merged**